File: 1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.py
class Solution:
    def restoreMatrix(self, rowSum: List[int], colSum: List[int]) -> List[List[int]]:
        rows, cols, r, c = len(rowSum), len(colSum), 0, 0
        mat = [[0 for _ in range(0, cols)] for _ in range(0, rows)]
        while r < rows and c < cols:
            mat[r][c] = min(rowSum[r], colSum[c])
            rowSum[r] -= mat[r][c]
            colSum[c] -= mat[r][c]
            if rowSum[r] == 0:
                r += 1
            if colSum[c] == 0:
                c += 1
        return mat

File: 1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.ts
function restoreMatrix(rowSum: number[], colSum: number[]): number[][] {
    const rows = rowSum.length
    const cols = colSum.length
    const mat: number[][] = Array(rows).fill(0).map(() => Array(cols).fill(0))
    for (let r=0; r < rows; r++) {
        for (let c=0; c < cols; c++) {
            mat[r][c] = Math.min(rowSum[r], colSum[c])
            rowSum[r] -= mat[r][c]
            colSum[c] -= mat[r][c]
        }
    }
    return mat
};